src/actions/sources/select.js

```diff
@@ -32,14 +32,30 @@ import {
   getSelectedLocation
 } from "../../selectors";
 
-import type { Location } from "../../types";
-import type { Action, ThunkArgs } from "../types";
+import type { Location, Source } from "../../types";
+import type { ThunkArgs } from "../types";
 
 declare type SelectSourceOptions = {
   tabIndex?: number,
   location?: { line: number, column?: ?number }
 };
 
+export const setSelectedLocation = (source: Source, location: Location) => ({
+  type: "SET_SELECTED_LOCATION",
+  source,
+  location
+});
+
+export const setPendingSelectedLocation = (url: string, options: Object) => ({
+  type: "SET_PENDING_SELECTED_LOCATION",
+  url: url,
+  line: options.location ? options.location.line : null
+});
+
+export const clearSelectedLocation = () => ({
+  type: "CLEAR_SELECTED_LOCATION"
+});
```

Comment:
I think i might be old school, but it would be nice for these to still be functions:

```js
export function clearSelectedLocation() {
  return {type: "CLEAR_SELECTED_LOCATION"}
}
```

this is pure style though

Reply:
hmm, what do you mean?

Reply:
> hmm, what do you mean?

Sounds like @lukaszsobek is talking about `this` context and how `function() {}` produces its own.